### PR TITLE
[TACHYON-1092] Improve creation of test conf in LocalTachyonCluster

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-1092

This is a followup PR to cleanup LocalTachyonCluster. This PR aims to fix the problem that in integration tests, the test conf properties set by a user might be still overwritten inside `LocalTachyonCluster` before `start`. After this PR, integration tests that wants to customize conf can `newTestConf` then `start` with this newly created conf object.

### CODE REVIEW

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by apc999** [comment]
@calvinjia , this PR addresses your comment in #1713 on properties being overwritten inside integration test.

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java [hunk 1]
```diff
@@ -134,83 +131,84 @@ public NetAddress getWorkerAddress() {
...
+    testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, 0),
         ramdiskPath);
-    mTestConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT, 0),
+    testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT, 0),
         Long.toString(mWorkerCapacityBytes));
-    UnderFileSystemUtils.mkdirIfNotExists(ramdiskPath, mTestConf);
+    UnderFileSystemUtils.mkdirIfNotExists(ramdiskPath, testConf);
 
-    int maxLevel = mTestConf.getInt(Constants.WORKER_TIERED_STORAGE_LEVEL_MAX);
+    int maxLevel = testConf.getInt(Constants.WORKER_TIERED_STORAGE_LEVEL_MAX);
```
Comment (calvinjia) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: This block of logic doesn't belong here right?

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java [hunk 2]
```diff
@@ -134,83 +131,84 @@ public NetAddress getWorkerAddress() {
...
 
     // Perform immediate shutdown of data server. Graceful shutdown is unnecessary and slow
-    mTestConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_QUIET_PERIOD, Integer.toString(0));
-    mTestConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT, Integer.toString(0));
+    testConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_QUIET_PERIOD, Integer.toString(0));
+    testConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT, Integer.toString(0));
 
     // Delete the temp dir by ufs, otherwise, permission problem may be encountered.
-    UnderFileSystemUtils.deleteDir(mTachyonHome, mTestConf);
+    UnderFileSystemUtils.deleteDir(mTachyonHome, testConf);
```
Comment (calvinjia) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: This logic doesn't belong here?

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java (hunk 1 above)
Comment (apc999) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: Agree with you that this is a little wierd. I will refactor this part.

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java (hunk 2 above)
Comment (apc999) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: Will be refactored.

**Comment by apc999** [comment]
@calvinjia  addressed your comment. PTAL

**Comment by calvinjia** [comment]
@apc999 Are the test failures related?

**Comment by apc999** [comment]
@calvinjia  I think so. Let me investigate them

**Comment by apc999** [comment]
@calvinjia I fixed the failing tests. I will create followup JIRAs to fully refactor `LocalTachyonCluster` and `LocalTachyonClusterMultiMaster`

**Comment by yupeng9** [comment]
LGTM

**Comment by gpang** [comment]
LGTM

**Pull request merged**